pkg/cli/cmd/directory/directory.go

```diff
@@ -21,3 +30,33 @@ type DirectoryCmd struct {
 	Check          CheckCmd          `cmd:"" help:"check" group:"directory"`
 	GetGraph       GetGraphCmd       `cmd:"" help:"get relation graph" group:"directory"`
 }
+
+type Message[T any] interface {
+	proto.Message
+	*T
+}
+
+func UnmarshalRequest[T any, M Message[T]](src string, msg M) error {
+	if src == "-" {
+		decoder := json.NewDecoder(os.Stdin)
+
+		err := decoder.Decode(msg)
```

Comment:
Shouldn't this path use `protojson.Unmarshal` too?

I think the function can be reorganized to first create an appropriate `io.Reader` based on the options (stdin, file, or string) and then use protojson to unmarshal from that reader.

Reply:
resolved, using bufio to read from stdin until EOF provided